lib/puppet/pops/loaders.rb

```diff
@@ -325,6 +326,18 @@ def instantiate_definition(definition, loader)
     end
   end
 
+  # Get the hierarchy of loaders from the top down (to enable testing).
+  #
+  def loader_hierarchy
+    loaders = []
+    loader = private_environment_loader
+    while loader
+      loaders << loader
+      loader = loader.parent
+    end
+    loaders.reverse
+  end
+
```

Comment:
probably better to make this a helper in the test so we don't expose a new method